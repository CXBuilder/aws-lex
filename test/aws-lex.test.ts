import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { YesNoBot } from './YesNoBot';

// example test. To run these tests, uncomment this file along with the
// example resource in lib/aws-lex-stack.ts
test('Bot Created', () => {
  const app = new cdk.App();
  // WHEN
  const stack = new cdk.Stack(app, 'MyTestStack');
  new YesNoBot(stack, 'YesNo', {
    name: 'cxbuilder-yes-no',
  });

  // THEN
  const template = Template.fromStack(stack);

  template.hasResourceProperties('AWS::Lex::CfnBot', {
    VisibilityTimeout: 300,
  });
});
