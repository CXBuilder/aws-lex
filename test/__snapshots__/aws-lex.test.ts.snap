// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YesNoBot matches snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "YesNoAliasD95BE263": {
      "Properties": {
        "BotAliasLocaleSettings": [
          {
            "BotAliasLocaleSetting": {
              "Enabled": true,
            },
            "LocaleId": "en_US",
          },
          {
            "BotAliasLocaleSetting": {
              "Enabled": true,
            },
            "LocaleId": "es_US",
          },
        ],
        "BotAliasName": "live",
        "BotId": {
          "Fn::GetAtt": [
            "YesNoBotB35E0693",
            "Id",
          ],
        },
        "BotVersion": {
          "Fn::GetAtt": [
            "YesNoVersion21018777070AD6092E",
            "BotVersion",
          ],
        },
      },
      "Type": "AWS::Lex::BotAlias",
    },
    "YesNoBotB35E0693": {
      "Properties": {
        "AutoBuildBotLocales": false,
        "BotLocales": [
          {
            "Intents": [
              {
                "DialogCodeHook": {
                  "Enabled": false,
                },
                "FulfillmentCodeHook": {
                  "Enabled": false,
                },
                "Name": "Yes",
                "SampleUtterances": [
                  {
                    "Utterance": "yeah",
                  },
                  {
                    "Utterance": "yep",
                  },
                  {
                    "Utterance": "yea",
                  },
                  {
                    "Utterance": "yes",
                  },
                  {
                    "Utterance": "all right",
                  },
                  {
                    "Utterance": "surely",
                  },
                  {
                    "Utterance": "yes sir",
                  },
                  {
                    "Utterance": "of course",
                  },
                  {
                    "Utterance": "absolutely",
                  },
                  {
                    "Utterance": "for sure",
                  },
                  {
                    "Utterance": "totally",
                  },
                  {
                    "Utterance": "correct",
                  },
                  {
                    "Utterance": "si",
                  },
                ],
              },
              {
                "DialogCodeHook": {
                  "Enabled": false,
                },
                "FulfillmentCodeHook": {
                  "Enabled": false,
                },
                "Name": "No",
                "SampleUtterances": [
                  {
                    "Utterance": "nope",
                  },
                  {
                    "Utterance": "nah",
                  },
                  {
                    "Utterance": "no",
                  },
                  {
                    "Utterance": "never",
                  },
                  {
                    "Utterance": "no thanks",
                  },
                  {
                    "Utterance": "no way",
                  },
                  {
                    "Utterance": "absolutely not",
                  },
                  {
                    "Utterance": "no thank you",
                  },
                  {
                    "Utterance": "dont",
                  },
                ],
              },
              {
                "DialogCodeHook": {
                  "Enabled": false,
                },
                "FulfillmentCodeHook": {
                  "Enabled": false,
                },
                "Name": "FallbackIntent",
                "ParentIntentSignature": "AMAZON.FallbackIntent",
              },
            ],
            "LocaleId": "en_US",
            "NluConfidenceThreshold": 0.4,
            "SlotTypes": [],
            "VoiceSettings": {
              "Engine": "neural",
              "VoiceId": "Joanna",
            },
          },
          {
            "Intents": [
              {
                "DialogCodeHook": {
                  "Enabled": false,
                },
                "FulfillmentCodeHook": {
                  "Enabled": false,
                },
                "Name": "Yes",
                "SampleUtterances": [
                  {
                    "Utterance": "sí",
                  },
                  {
                    "Utterance": "claro",
                  },
                  {
                    "Utterance": "por supuesto",
                  },
                  {
                    "Utterance": "afirmativo",
                  },
                  {
                    "Utterance": "correcto",
                  },
                  {
                    "Utterance": "exacto",
                  },
                  {
                    "Utterance": "efectivamente",
                  },
                  {
                    "Utterance": "cierto",
                  },
                  {
                    "Utterance": "desde luego",
                  },
                  {
                    "Utterance": "así es",
                  },
                  {
                    "Utterance": "seguro",
                  },
                  {
                    "Utterance": "vale",
                  },
                  {
                    "Utterance": "bueno",
                  },
                  {
                    "Utterance": "de acuerdo",
                  },
                  {
                    "Utterance": "está bien",
                  },
                ],
              },
              {
                "DialogCodeHook": {
                  "Enabled": false,
                },
                "FulfillmentCodeHook": {
                  "Enabled": false,
                },
                "Name": "No",
                "SampleUtterances": [
                  {
                    "Utterance": "no",
                  },
                  {
                    "Utterance": "para nada",
                  },
                  {
                    "Utterance": "de ninguna manera",
                  },
                  {
                    "Utterance": "negativo",
                  },
                  {
                    "Utterance": "nunca",
                  },
                  {
                    "Utterance": "jamás",
                  },
                  {
                    "Utterance": "no gracias",
                  },
                  {
                    "Utterance": "en absoluto",
                  },
                  {
                    "Utterance": "de ningún modo",
                  },
                  {
                    "Utterance": "no quiero",
                  },
                  {
                    "Utterance": "no me interesa",
                  },
                ],
              },
              {
                "DialogCodeHook": {
                  "Enabled": false,
                },
                "FulfillmentCodeHook": {
                  "Enabled": false,
                },
                "Name": "FallbackIntent",
                "ParentIntentSignature": "AMAZON.FallbackIntent",
              },
            ],
            "LocaleId": "es_US",
            "NluConfidenceThreshold": 0.4,
            "SlotTypes": [],
            "VoiceSettings": {
              "Engine": "neural",
              "VoiceId": "Lupe",
            },
          },
        ],
        "DataPrivacy": {
          "ChildDirected": false,
        },
        "IdleSessionTTLInSeconds": 300,
        "Name": "cxbuilder-yes-no",
        "RoleArn": {
          "Fn::GetAtt": [
            "YesNoRole10D00A8B",
            "Arn",
          ],
        },
        "TestBotAliasSettings": {
          "BotAliasLocaleSettings": [
            {
              "BotAliasLocaleSetting": {
                "Enabled": true,
              },
              "LocaleId": "en_US",
            },
            {
              "BotAliasLocaleSetting": {
                "Enabled": true,
              },
              "LocaleId": "es_US",
            },
          ],
        },
      },
      "Type": "AWS::Lex::Bot",
    },
    "YesNoRole10D00A8B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lexv2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "YesNoRoleLexPolicy8BF6BC9D": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "polly:SynthesizeSpeech",
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:logs:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:*:log-stream:*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "YesNoRoleLexPolicy8BF6BC9D",
        "Roles": [
          {
            "Ref": "YesNoRole10D00A8B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "YesNoVersion21018777070AD6092E": {
      "Properties": {
        "BotId": {
          "Fn::GetAtt": [
            "YesNoBotB35E0693",
            "Id",
          ],
        },
        "BotVersionLocaleSpecification": [
          {
            "BotVersionLocaleDetails": {
              "SourceBotVersion": "DRAFT",
            },
            "LocaleId": "en_US",
          },
          {
            "BotVersionLocaleDetails": {
              "SourceBotVersion": "DRAFT",
            },
            "LocaleId": "es_US",
          },
        ],
      },
      "Type": "AWS::Lex::BotVersion",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
